app/models/seance.rb

```diff
@@ -1,7 +1,21 @@
 class Seance < ApplicationRecord
   belongs_to :hall
   belongs_to :movie
+  has_many :reservations
 
   validates :movie_id, presence: true
   validates :hall_id, presence: true
+
+  def available_seats
+    all_seats = hall.generate_seats
+    taken = taken_seats
+
+    all_seats.reject { |seat| taken.include? seat }
```

Comment:
Ruby power: `all_seats - taken`

Reply:
Wow! It's sounds like... Ruby 🤣 Thank you!